syncer/syncer_test.go

```diff
@@ -1112,3 +1117,59 @@ func TestGetATXsOldAndCurrentEpoch(t *testing.T) {
 	assert.NoError(t, syncer.getATXs(context.TODO(), types.NewLayerID(8)))
 	assert.Equal(t, uint32(4), atomic.LoadUint32(&mf.atxsCalls))
 }
+
+func TestSyncMissingLayer(t *testing.T) {
+	lg := logtest.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	ticker := newMockLayerTicker()
+	fetcher := mocks.NewMocklayerFetcher(ctrl)
+	fetcher.EXPECT().GetEpochATXs(gomock.Any(), gomock.Any()).AnyTimes()
+
+	genesis := types.GetEffectiveGenesis()
+	last := genesis.Add(4)
+	failed := last.Sub(1)
+
+	memdb := mesh.NewMemMeshDB(lg.WithName("meshDB"))
+	atxStore := database.NewMemDatabase()
+	goldenATXID := types.ATXID(types.HexToHash32("77777"))
+	atxdb := activation.NewDB(atxStore, nil,
+		activation.NewIdentityStore(database.NewMemDatabase()),
+		layersPerEpoch, goldenATXID, nil, lg.WithName("atxDB"))
+	svmstate := mmocks.NewMockstate(gomock.NewController(t))
+	svmstate.EXPECT().GetStateRoot().AnyTimes()
+
+	tortoise := mmocks.NewMocktortoise(ctrl)
+
+	m := mesh.NewMesh(memdb, atxdb, tortoise, nil, svmstate, lg.WithName("mesh"))
```

Comment:
maybe a mock mesh is better in syncer. the test setup was mostly copied from the old syncer code. i didn't do a good job updating it.

Reply:
you meant that it will be better to replace mesh with interface and use mock for it? i guess it will be cleaner, on the other hand the interaction between syncer and mesh got tested.